.github/workflows/code-climate.yaml

```
@@ -0,0 +1,62 @@
+name: Generate, Aggregate and Upload coverage reports to Code Climate
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  api:
+    env:
+      CC_TEST_REPORTER_ID: 861e868a6d2776ca55c53ef1ee5042f1de0db45137b89eeeda67e15549bfcdbf
+
+    name: Create, format and upload Unit Test Coverage
+    runs-on: ubuntu-latest
+    steps:
+
+      # Checkout CIRMO-DPIA Github Repository
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      # Get Code Climate binary
+      - name: Download Code Climate Binary
+        run: |
+          curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+
+      # Permissions applied to the Code Climate Executable
+      - name: Apply executable perms to Code Climate Binary
+        run: |
+          chmod +x ./cc-test-reporter
+      
+      # Before build prep
+      - name: Before build
+        run: ./cc-test-reporter before-build
+
+      # Set required Git env vars for either pull request
+      - name: Set ENV for codeclimate
+        run: |
+          echo "GIT_BRANCH=${{ github.event.pull_request.head.ref }}" >> $GITHUB_ENV
+          echo "GIT_COMMIT_SHA=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+
+      # Trimming the ref to main in order to publish correct report (paambaati)
+      - name: Set ref/head/main to main
+        run: |
+          echo "GIT_BRANCH=main" >> $GITHUB_ENV
+        if: env.GIT_BRANCH == 'refs/heads/main'
+
```

@akroon3r  - Github workflow is failing. You might be missing env setup step here

```suggestion

      # Setup environment variables
      - name: Setup environment
        run: |
          make api-setup-env

```